### PR TITLE
Support mprotect test to run with nvdimm device

### PR DESCRIPTION
**Pull request by harish-24**
1. mprotect: Add backing file as an input
Patch adds backing file as an input for mprotect test

2. Add support to create and run test on nvdimm device
Patch adds support to create and run on nvdimm device

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
``` 
 (1/2) memory/mprotect.py:Mprotect.test;back_file-nr_pages-positive-c5b1:  PASS (95.10 s)
 (2/2) memory/mprotect.py:Mprotect.test;back_file-nr_pages-negative-65ab:  PASS (94.41 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 189.78 s
```

File: memory/mprotect.py [hunk 1]
```diff
@@ -37,20 +38,59 @@ def copyutil(self, file_name):
...
+        self.file_type = self.params.get('file_type', default="block")
         self.failure = self.params.get('failure', default=False)
 
         if not self.nr_pages:
             memsize = int(memory.meminfo.MemFree.b * 0.9)
             self.nr_pages = memsize / memory.get_page_size()
 
-        for package in ['gcc', 'make']:
+        deps = ['gcc', 'make']
+
```
Comment (PraveenPenguin) [inline memory/mprotect.py:82]: please remove this white space

File: memory/mprotect.py [hunk 2]
```diff
@@ -37,20 +38,59 @@ def copyutil(self, file_name):
...
+        if self.file_type == 'nvdimm':
+            deps.extend(['ndctl'])
+            if distro.detect().name == 'rhel':
+                deps.extend(['daxctl'])
+
```
Comment (PraveenPenguin) [inline memory/mprotect.py:87]: this as well

File: memory/mprotect.py.data/mprotect.c [hunk 3]
```diff
@@ -43,13 +43,30 @@ int main(int argc, char **argv)
...
+	char *filepath;
 
-	if (argc == 3)
+	if (argc > 2)
 		induce = atoi(argv[2]);
+	if (argc > 3)
+		filepath = argv[3];
 
-	fd = open("/dev/zero", O_RDWR);
+	fd = open(filepath, O_CREAT | O_RDWR);
```
Comment (PraveenPenguin) [inline memory/mprotect.py.data/mprotect.c:53]: :+1:

**Comment by harish-24** [comment]
@PraveenPenguin Pushed with changes.

**Pull request merged**